app/src/main/java/com/example/kharrison/my_application/CalculatorController.kt

```diff
@@ -0,0 +1,162 @@
+         five.setOnClickListener { clickNumber("5") }
+         six.setOnClickListener { clickNumber("6") }
+         seven.setOnClickListener { clickNumber("7") }
+         eight.setOnClickListener { clickNumber("8") }
+         nine.setOnClickListener { clickNumber("9") }
+         zero.setOnClickListener { clickNumber("0") }
+         add.setOnClickListener { clickOperator(ADDITION) }
+         subtract.setOnClickListener { clickOperator(SUBTRACTION) }
+         multiply.setOnClickListener { clickOperator(MULTIPLICATION) }
+         divide.setOnClickListener { clickOperator(DIVISION) }
+         equals.setOnClickListener { calculate() }
+         clear.setOnClickListener { clearInput() }
+      }
+      invalidateControls()
+   }
+
+   private fun invalidateControls() {
+      val isEnabled = (!leftHandSide.isEmpty() || result != 0) && !isOperationSelected
+
+      view.add.isClickable = isEnabled
+      view.subtract.isClickable = isEnabled
+      view.divide.isClickable = isEnabled
+      view.multiply.isClickable = isEnabled
+   }
+
+   private fun updateText() {
+      view.input.text = inputString
+      view.answer.text = outputString
+   }
+
+   private fun updateOutPutString() {
+      leftHandSide = ""
+      rightHandSide = ""
+      inputString = ""
+      updateText()
+   }
+
+   private fun addToInputString(input : String) {
+      inputString += input
+      updateText()
+   }
+
+   private fun clickNumber(num : String) {
+      if (!isOperationSelected) {
+         leftHandSide += num
+      } else {
+         rightHandSide += num
+      }
+      addToInputString(num)
+      invalidateControls()
+   }
+
+   private fun clickOperator(operator : Int) {
+      isOperationSelected = true
+      when (operator) {
+         ADDITION -> {
+            selectedOperator = ADDITION
+            addToInputString(" + ")
+         }
+         SUBTRACTION -> {
+            selectedOperator = SUBTRACTION
+            addToInputString(" - ")
+         }
+         MULTIPLICATION -> {
+            selectedOperator = MULTIPLICATION
+            addToInputString(" * ")
+         }
+         DIVISION -> {
+            selectedOperator = DIVISION
+            addToInputString(" / ")
+         }
+      }
+      invalidateControls()
+   }
+
+   private fun calculate() {
+      isOperationSelected = false
+      val leftNum = if (leftHandSide.isEmpty()) result else leftHandSide.toInt()
+      val rightNum = if (rightHandSide.isEmpty()) result else rightHandSide.toInt()
+      when (selectedOperator) {
+         ADDITION -> add(leftNum, rightNum)
+         SUBTRACTION -> subtract(leftNum, rightNum)
+         MULTIPLICATION -> multiply(leftNum, rightNum)
+         DIVISION -> division(leftNum, rightNum)
+      }
+      invalidateControls()
+   }
+
+   private fun add(leftNum : Int, rightNum : Int) {
+      result = mathHelper.add(leftNum, rightNum)
+      outputString = result.toString()
+      updateOutPutString()
+   }
```

Comment:
There's a lot of repeated code in `add`, `subtract`, etc. You might consider moving the common code out of the individual functions. 

    when (selectedOperator) {
         ADDITION -> restult = mathHelper.add(leftNum, rightNum)
         ...
    }
    
    outputString = result.toString()
    updateOutPutString()

    ....